datalad_gooey/dataladcmd_exec.py

```diff
@@ -62,53 +67,71 @@ def execute(self, cmd: str,
         self._futures.add(self._threadpool.submit(
             self._cmdexec_thread,
             cmd,
-            **kwargs
+            kwargs,
+            exec_params,
         ))
 
-    def _cmdexec_thread(self, cmdname, **kwargs):
+    def _cmdexec_thread(self, cmdname, cmdkwargs, exec_params):
         """The code is executed in a worker thread"""
-        print('EXECINTHREAD', cmdname, kwargs)
+        print('EXECINTHREAD', cmdname, cmdkwargs, exec_params)
+        preferred_result_interval = exec_params.get(
+            'preferred_result_interval', 1.0)
+        res_override = exec_params.get(
+            'result_override', {})
         # get_ident() is an int, but in the future we might want to move
         # to PY3.8+ native thread IDs, so let's go with a string identifier
         # right away
         thread_id = str(threading.get_ident())
         self.execution_started.emit(
             thread_id,
             cmdname,
-            kwargs,
+            cmdkwargs,
         )
         # get functor to execute, resolve name against full API
         cmd = getattr(dlapi, cmdname)
+        cls = get_wrapped_class(cmd)
 
         # enforce return_type='generator' to get the most responsive
         # any command could be
-        kwargs['return_type'] = 'generator'
+        cmdkwargs['return_type'] = 'generator'
         # Unless explicitly specified, force result records instead of the
         # command's default transformation which might give Dataset instances
         # for example.
-        if 'result_xfm' not in kwargs:
-            kwargs['result_xfm'] = None
+        if 'result_xfm' not in cmdkwargs:
+            cmdkwargs['result_xfm'] = None
 
-        if 'dataset' in kwargs:
+        if 'dataset' in cmdkwargs:
             # Pass actual instance, to have path arguments resolved against it
             # instead of Gooey's CWD.
-            kwargs['dataset'] = dlapi.Dataset(kwargs['dataset'])
+            cmdkwargs['dataset'] = dlapi.Dataset(cmdkwargs['dataset'])
+        gathered_results = []
+        last_report_ts = time()
         try:
-            for res in cmd(**kwargs):
-                self.result_received.emit(res)
+            for res in cmd(**cmdkwargs):
+                t = time()
+                res.update(res_override)
+                gathered_results.append(res)
+                if (t - last_report_ts) > preferred_result_interval:
```

Comment:
May be `and len(gathered_results) > 0` ?
If there was nothing yet, we'd be sending a signal that says: "nothing happened yet" which is kinda the opposite of what a signal is for.